src/crud-methods.js

```diff
@@ -0,0 +1,160 @@
+/* eslint-disable */
+import { setStorage } from './storage.js';
+import { clearAll } from './clear.js';
+
+
+export { deleteTask, updateStorage, tasks, addTask };
+
+const toDoList = document.getElementById('to-dos');
+const clearBtn = document.getElementById('clear-button');
+
+class Task {
+  constructor(description, completed, index) {
+    this.description = description;
+    this.completed = completed;
+    this.index = index;
+  }
+}
+
+let tasks = [];
+const addTask = (taskValue) => {
+  const localData = JSON.parse(localStorage.getItem('todo'));
+  localData.map((i) => tasks.push(i));
+  const taskContainer = document.createElement('div');
+  taskContainer.className = 'list-item';
+  taskContainer.innerHTML += `
+    <input type="checkbox" class="checkbox">
+    <span>${taskValue}</span>
+    <i class="fas fa-ellipsis-vertical"></i>
+    <i class="fas fa-trash-can"></i>
+  `;
+  toDoList.appendChild(taskContainer);
+
+  const checkbox = document.querySelectorAll('.checkbox');
+  checkbox.forEach((i) => {
+    i.addEventListener('change', () => {
+      i.parentElement.classList.toggle('list-selected-item');
+      i.nextElementSibling.classList.toggle('text-selected');
+      i.parentElement.lastElementChild.classList.toggle('delete-icon');
+      i.parentElement.lastElementChild.previousElementSibling.classList.toggle('disable-edit');
+      updateStorage();
+    });
+  });
+
+  const newTask = new Task(taskValue, false, checkbox.length);
+  tasks.push(newTask);
+  setStorage(tasks);
+
+  const edit = document.querySelectorAll('.fa-ellipsis-vertical');
+  edit.forEach((i) => {
+    i.addEventListener('click', () => {
+      editTask(taskContainer, i.previousElementSibling);
+      i.parentElement.classList.add('list-selected-item');
+    });
+  });
+
+  const remove = document.querySelectorAll('.fa-trash-can');
```

Comment:
It is a good coding practice to put all declarations at the top of each script because it gives cleaner code and reduces the possibility of unwanted re-declarations.

Reply:
Thank you for pointing that out.